tests/sentry/tasks/test_sentry_apps.py

```diff
@@ -770,3 +790,141 @@ def test_saves_error_event_id_if_in_header(self, safe_urlopen):
         assert first_request["organization_id"] == self.install.organization_id
         assert first_request["error_id"] == "d5111da2c28645c5889d072017e3445d"
         assert first_request["project_id"] == "1"
+
+    @with_feature("organizations:disable-sentryapps-on-broken")
+    @patch("sentry.utils.sentry_apps.webhooks.safe_urlopen", side_effect=Timeout)
+    def test_does_not_raise_error_if_unpublished(self, safe_urlopen):
+        """
+        Tests that buffer records when unpublished app has a timeout but error is not raised
+        """
+        self.sentry_app.update(status=SentryAppStatus.UNPUBLISHED)
+        events = len(self.sentry_app.events)
+        data = {"issue": serialize(self.issue)}
+        # we don't raise errors for unpublished and internal apps
+        send_webhooks(installation=self.install, event="issue.assigned", data=data, actor=self.user)
+
+        requests = self.buffer.get_requests()
+        requests_count = len(requests)
+
+        assert safe_urlopen.called
+        assert requests_count == 1
```

Comment:
nit: you can probably don't need the `requests_count` variable here, just inline it in the assert.